src/OrleansRuntime/Streams/PersistentStream/PersistentStreamPullingAgent.cs

```diff
@@ -509,6 +509,10 @@ private void StartInactiveCursors(StreamConsumerCollection streamData)
                     }
                     catch (Exception exc)
                     {
+                        if (consumerData.Cursor != null)
+                        {
+                            consumerData.Cursor.RecordDeliveryFailure();
```

Comment:
That was the first place I looked to add this, but the error protocol, and the block that calls it, both handle too many error cases.  I only want to mark this as undeliverable if we actually tried to deliver it and failed.  This block is the only place that handles that specific error alone.